src/Util/Common.hs

```diff
@@ -58,4 +60,54 @@ gameObjects game = map playerObject (gamePlayers game)
 
 
 updateObjects :: Game -> Game
-updateObjects game = foldr (\obj -> objectOnUpdate obj $ obj) game (gameObjects game)
\ No newline at end of file
+updateObjects game = foldr (\obj -> stringToUpdateFunction (objectOnUpdateName obj) $ obj) game (gameObjects game)
+
+activatePlayer :: String -> Object -> Game -> Game
+activatePlayer name object game =
+  foldr (\player -> (stringToActivateFunction (objectOnActivateName . playerObject $ player))
+                    (objectsCollide object (playerObject player))
+                    (playerObject player)
+        )
+        game
+        (filter isTarget (gamePlayers game))
+  where isTarget player = (objectName . playerObject $ player) == name
+
+namedUpdateFunctions :: [NamedUpdateFunction]
+namedUpdateFunctions = [("onUpdateMario", activatePlayer "luigi")]
+
+namedActivateFunctions :: [NamedActivateFunction]
+namedActivateFunctions = [("resizeSelf", resizeSelf)]
+
+lookUpWithDefaultFunction :: v -> [(k, v)] -> v
```

Comment:
Rename to `lookUpWithDefault` 
lookUpWithDefault should take 3 arguments: name :: k, defaultValue :: v and a list of tuples :: [(k, v)]
it should search a list for the pair (name, value) and return the value
if there is no such pair in the list,  then it should return defaultValue